PennMobile/Dining/SwiftUI/Views/Venue/DiningVenueView.swift

```diff
@@ -73,7 +71,9 @@ struct CustomHeader: View {
                         showMissingDiningTokenAlert = true
                         return
                     }
-                    DiningViewModelSwiftUI.instance.refreshBalance()
+                    Task.init() {
+                        await DiningViewModelSwiftUI.instance.refreshBalance()
+                    }
```

Comment:
You can add a `.task` modifier at the end of the view to trigger a async function. More details can be found [here](https://developer.apple.com/documentation/swiftui/view/task(priority:_:))